src/features/patients/pages/PatientForm.tsx

```diff
@@ -126,13 +176,22 @@ export const PatientForm: FC = () => {
 
       {!patientSurveySchemaIsLoading ? (
         <JsonSchemaForm
+          onSubmit={handleSubmit}
           id="patientSurvey"
           jsonSchema={patientSurveySchema?.jsonSchema ?? {}}
           uiSchema={patientSurveySchema?.uiSchema ?? {}}
           onChange={handleSurveyChange}
           refresh={refreshForms}
         >
-          <Button variant="contained" color="primary" onClick={handleSubmit}>
+          <Button
+            variant="contained"
+            color="primary"
+            type="submit"
+            onClick={() => {
+              buttonRef.current?.click();
+              handleSubmit();
+            }}
```

Comment:
technically this is triggering a submit event on the survey form, then on the patient form, and then handling the submit...